src/Runner.Worker/ActionManager.cs

```diff
@@ -571,34 +625,55 @@ private async Task DownloadRepositoryActionAsync(IExecutionContext executionCont
                                 }
 
                                 httpClient.DefaultRequestHeaders.UserAgent.AddRange(HostContext.UserAgents);
-                                using (var result = await httpClient.GetStreamAsync(archiveLink))
+                                using (var response = await httpClient.GetAsync(link))
                                 {
-                                    await result.CopyToAsync(fs, _defaultCopyBufferSize, actionDownloadCancellation.Token);
-                                    await fs.FlushAsync(actionDownloadCancellation.Token);
-
-                                    // download succeed, break out the retry loop.
-                                    break;
+                                    if (response.IsSuccessStatusCode)
+                                    {
+                                        using (var result = await httpClient.GetStreamAsync(link))
```

Comment:
why we need to GetAsync() and GetStreamAsync()? looks like we are making the same request twice?

Reply:
Yep, just fixed that, it should be `response.Content.ReadAsStreamAsync()`